make test fails on macOS Catalina: Library not loaded: @rpath/libmysqlclient.21.dylib

I am on macOS 10.15.3, perlbrew with perl version 5.30. I first installed mysql from

https://dev.mysql.com/downloads/mysql/

Downloaded disk image file:  mysql-8.0.19-macos10.15-x84_64.dmg
After the installation I added to my `~/.zshrc` 

```
 export PATH="$PATH:/usr/local/mysql-8.0.19-macos10.15-x86_64/bin"
export LD_LIBRARY_PATH=/usr/local/mysql-8.0.19-macos10.15-x86_64/lib
export DYLD_LIBRARY_PATH=/usr/local/mysql-8.0.19-macos10.15-x86_64/lib
```

Then I ran `perl Makefile.PL; make; make test`:

```
"/Users/hakonhaegland/perl5/perlbrew/perls/perl-5.30.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- mysql.bs blib/arch/auto/DBD/mysql/mysql.bs 644
PERL_DL_NONLAZY=1 "/Users/hakonhaegland/perl5/perlbrew/perls/perl-5.30.1/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00base.t .............................. 1/6
#   Failed test 'use DBD::mysql;'
#   at t/00base.t line 15.
#     Tried to use 'DBD::mysql'.
#     Error:  Can't load '/Users/hakonhaegland/perl/DBD-mysql-4.050/blib/arch/auto/DBD/mysql/mysql.bundle' for module DBD::mysql: dlopen(/Users/hakonhaegland/perl/DBD-mysql-4.050/blib/arch/auto/DBD/mysql/mysql.bundle, 2): Library not loaded: @rpath/libmysqlclient.21.dylib
#   Referenced from: /Users/hakonhaegland/perl/DBD-mysql-4.050/blib/arch/auto/DBD/mysql/mysql.bundle
#   Reason: image not found at /Users/hakonhaegland/perl5/perlbrew/perls/perl-5.30.1/lib/5.30.1/darwin-2level/DynaLoader.pm line 197.
#  at t/00base.t line 15.
# Compilation failed in require at t/00base.t line 15.
# BEGIN failed--compilation aborted at t/00base.t line 15.
Bailout called.  Further testing stopped:  Unable to load DBD::mysql
FAILED--Further testing stopped: Unable to load DBD::mysql
make: *** [test_dynamic] Error 255
```

It is strange that it cannot find `libmysqlclient.21.dylib` since I set `LD_LIBRARY_PATH` and the directory indeed has the library:

```
$ ls /usr/local/mysql-8.0.19-macos10.15-x86_64/lib
libcrypto.1.1.dylib          libmysqlclient.a             libprotobuf-lite.3.6.1.dylib libprotobuf.dylib            mecab
libcrypto.dylib              libmysqlclient.dylib         libprotobuf-lite.dylib       libssl.1.1.dylib             pkgconfig
libmysqlclient.21.dylib      libmysqlservices.a           libprotobuf.3.6.1.dylib      libssl.dylib                 plugin
```

--- a/lib/DBD/mysql/INSTALL.pod
+++ b/lib/DBD/mysql/INSTALL.pod
@@ -542,7 +542,14 @@ mysql client libs. The easiest way to install these is using Homebrew
 
 Once you have Homebrew set up, you can simply install the dependencies using
 
-    brew install openssl mysql-connector-c
+    brew install openssl@3 mysql-client
+
+Then make sure to add the path for L<mysql_config> to your path:
+
+    export PATH=/usr/local/Cellar/mysql-client/8.1.0/bin:$PATH
+
+Make sure to use the correct path for your installation. Check the output of L<brew list mysql-client>
+to see the correct path. You can use L<~/.zshrc> to set the path for future sessions as well.
 
 Then you can install DBD::mysql using your cpan client.
 

